contracts/src/v0.8/vrf/dev/SubscriptionAPI.sol

```diff
@@ -379,13 +384,13 @@ abstract contract SubscriptionAPI is ConfirmedOwner, IERC677Receiver, IVRFSubscr
    * @inheritdoc IVRFSubscriptionV2Plus
    */
   function acceptSubscriptionOwnerTransfer(uint256 subId) external override nonReentrant {
-    if (s_subscriptionConfigs[subId].owner == address(0)) {
+    address oldOwner = s_subscriptionConfigs[subId].owner;
+    if (oldOwner == address(0)) {
```

Comment:
Let's say we define a constant value called `address ZERO_ADDRESS = address(0)` somewhere in some shared file (of course, that increases the contract size a bit). In that case, we replace each occurrence of `address(0)` with the constant's name. Now, this value is re-used within all function calls that require it and no temp var is created just for the sake of the condition check. From the Solidity perspective, would that approach be better or worse and why? Again, I'm just curious and trying to learn a thing or two about contract optimization techniques.

Reply:
Seems like there will be no difference as the compiler would replace each constant values with the actual values wherever it is used. Gas cost for `ISZERO` and `EQ` are both 3.

These questions are good. Keep them coming. I wouldn't say I'm an expert in contract optimisation. I'm just tweaking things around and see which actually saves gas. But I do love to optimise things.